### ISSUE TITLE
Assets are not present in iOS build.

### ISSUE THREAD

**Issue by rares-munteanu** [expo/eas-cli#836]
### Summary

Static assets are not present in the iOS build (images, animations, etc).

Command tried : 
 eas build --profile preview --platform ios 

Also generated the iOS folder with expo run:ios and then archived it with xcode but the assets are still not present.

The whole app is placed in the app folder having the assets folder in it. In app.json the "assetBundlePatterns" is still set to  ["**/*"]. But I also tried to modify it to specifically match the asset folder and all files inside it. The issue still persist.

An example of image importing : require('@myApp/assets/onboarding/onboarding3.png'). 
The project is setup with typescript template.

The tsconfig file setup the alias for myApp this way : 
```
{
   compilerOptions : {
       "paths": {
      "@myApp/*": ["app/*/index", "app/*"]
    },
   }
}
```
Also the babel.config.js file : 
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],

    plugins: [
      [
        'module-resolver',
        {
          alias: {
            '@myApp': './app',
          },
        },
      ],
    ],
  };
};

### Managed or bare?

Bare - with expo dev client.

### Environment

  EAS CLI 0.35.0 environment info:
    System:
      OS: macOS 11.6
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
      Yarn: 1.22.17 - /usr/local/bin/yarn
      npm: 8.1.0 - ~/.nvm/versions/node/v14.18.1/bin/npm
    Utilities:
      Git: 2.32.0 - /usr/local/bin/git
    npmPackages:
      expo: ^43.0.0 => 43.0.3 
      expo-dev-client: ~0.6.3 => 0.6.3 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      eas-cli: 0.35.0
      expo-cli: 4.13.0

### Reproducible demo or steps to reproduce from a blank project

[Snack minimal reproducible demo](https://snack.expo.dev/@irares/eas-bug---reproduce)

I didn't manage to setup snack to read the alias for the app folder but the structure is the same I have in my project. Simple image require. 

eas.json file : 

```
{
  "cli": {
    "version": ">= 0.35.0"
  },
  "build": {
    "development": {
      "distribution": "internal",
      "android": {
        "gradleCommand": ":app:assembleDebug"
      },
      "ios": {
        "buildConfiguration": "Debug"
      }
    },
    "preview": {
      "distribution": "internal"
    },
    "production": {}
  },
  "submit": {
    "production": {}
  }
}
```

**Comment by brentvatne** [comment]
please share a full repo that we can clone to reproduce.

**Comment by rares-munteanu** [comment]
Sure. I created a dummy project from scratch that replicate the environment of the actual project. 
https://github.com/rares-munteanu/EAS-cli-debug.git

If you run `eas build --profile preview --platform ios` the assets within the app/assets folder will not be present in the app.

**Comment by brentvatne** [comment]
the problem is that your metro.config.js is not set up properly for eas build. you are not extending the assetPlugins field from the default config

**Comment by rares-munteanu** [comment]
Thank you for your response. Although it seems there is no field called **assetPlugins** in the default config I managed to find a [proper way](https://github.com/Yonom/react-native-svg-transformer/commit/e56be27ee6bbef85f795d7921cf78f41fd906570) to set up metro.config.js.

**Comment by brentvatne** [comment]
> Thank you for your response. Although it seems there is no field called **assetPlugins** in the default config I managed to find a [proper way](https://github.com/Yonom/react-native-svg-transformer/commit/e56be27ee6bbef85f795d7921cf78f41fd906570) to set up metro.config.js.

https://facebook.github.io/metro/docs/configuration#assetplugins

**Comment by rares-munteanu** [comment]
I didn't search the metro docs for that plugin field. I looked into the source code of @expo/metro-config by tapping on types because you said I didn't extend the assetPlugin field from the default config which is returned by the getDefaultConfig function.

If we click on @expo/metro-config we have the getDefaultConfig function :
`export declare function getDefaultConfig(projectRoot: string, options?: DefaultConfigOptions): MetroConfig.InputConfigT;`

If we click on InputConfigT we get this : 
![image](https://user-images.githubusercontent.com/49690340/146332993-250d9da9-7798-4d48-b1d2-10b0a1e69a57.png)

Now, knowing from the metro docs that assetPlugins is a field present in transformer options we click on TransformerConfigT and we get this : 
![image](https://user-images.githubusercontent.com/49690340/146333291-3e86aadb-39d4-43f7-99cf-24728a62a6c2.png)

Then on GetTransformOptions : 
![image](https://user-images.githubusercontent.com/49690340/146333367-e1579fac-9c47-4f03-b009-71f8fac7ebfb.png)

And then on GetTransformOptionsOpts :
![image](https://user-images.githubusercontent.com/49690340/146333524-0ff3db57-7d71-4721-a890-99afa2437652.png)

That's why I said there is no field called that way. But maybe is set/imported somewhere else and I didn't catch it.

Anyway, the issue has been solved. Thank you again.
